src/main.py

```diff
@@ -1,125 +1,250 @@
 # Leah
+# main.py
 # Written by aquova et al., 2022
-# https://github.com/aquova/leah
+# https://github.com/StardewValleyDiscord/leah
 
+import random
+import strings
 import discord
-from config import client, DISCORD_KEY,\
-    ART_CHANS, MOD_CHANS, VERIFY_CHAN, GALLERY_CHAN, SHOWCASE_CHAN,\
-    SHOWCASE_ROLES
+from discord.ext import commands
+from config import COMMAND_PREFIX, EXTENSIONS, GUILDS, DISCORD_KEY, DISCORD_INTENTS, \
+    ART_CHANS, MOD_CHANS, VERIFY_CHAN, GALLERY_CHAN, SHOWCASE_CHAN, VERIFY_ROLES, SHOWCASE_ROLES, ADMIN_ROLE
+from importlib import reload
 from typing import Optional
+from utils import check_roles, format_roles_error
 
-class Leah:
+
+# Bot definition
+class Leah(commands.Bot):
     def __init__(self):
-        self.posted = set()
+        super().__init__(
+            command_prefix=COMMAND_PREFIX,
+            intents=DISCORD_INTENTS,
+            description=strings.get("client_description"),
+            allowed_mentions=discord.AllowedMentions.none())
+
+    async def sync_guild(self, guild: discord.Guild):
+        self.tree.copy_global_to(guild=guild)
+        await self.tree.sync(guild=guild)
+
+    async def setup_hook(self) -> None:
+        # Load all extensions on setup
+        for ext in EXTENSIONS:
+            await self.load_extension(name=ext)
+
+    async def on_ready(self):
+        print(strings.get("client_login").format(bot.user.name, bot.user.discriminator, bot.user.id))
 
-leah = Leah()
+    def reload_strings(self):
+        reload(strings)
 
-@client.event
-async def on_ready():
-    print("Logged in as:")
-    print(client.user.name)
-    print(client.user.id)
 
-@client.event
-async def on_reaction_add(reaction: discord.Reaction, user: discord.User):
-    # Ignore messages that have been handled previously
-    if reaction.message.id in leah.posted:
+bot = Leah()
+
+
+@bot.listen(name="on_message")
+async def on_message(message: discord.Message) -> None:
+    """
+    For messages in listening channels, reposts a preview to the verification channel.
+    """
+    # Ignore all bots
+    if message.author.bot:
         return
 
-    # Staff reactions to verification channel posts by this client
-    if reaction.message.channel.id == VERIFY_CHAN:
-        # Ignore reactions to posts not by this client in verification channel
-        if client.user != reaction.message.author:
-            return
+    # Send posts from listening channels to verification channel
+    if message.channel.id in ART_CHANS:
+        await verify_art(message=message)
+
+@bot.tree.context_menu(name=strings.get("app_name_publish"), guilds=GUILDS)
```

Comment:
Would we want to be able to change the command name on the fly? All the other commands are (I think) hardcoded.

Reply:
I didn't see a reason not to since we can sync apps and strings at any time, but we can remove it if you prefer